addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/OneWireGPIOSensorReader.java

```diff
@@ -0,0 +1,43 @@
+package org.openhab.binding.onewiregpio.internal;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OneWireGPIOSensorReader {
+
+    private static Logger logger = LoggerFactory.getLogger(OneWireGPIOSensorReader.class);
+
+    private static String FILE_TEMP_MARKER = "t=";
+
+    private static BigDecimal INVALID_TEMPERATURE = BigDecimal.valueOf(-127);
+
+    public static BigDecimal readSensorTemperature(String gpioFile) {
+
+        try (Stream<String> stream = Files.lines(Paths.get(gpioFile))) {
+            Optional<String> temperatureLine = stream.filter(s -> s.contains(FILE_TEMP_MARKER)).findFirst();
+            if (temperatureLine.isPresent()) {
+                String line = temperatureLine.get();
+                String tempString = line.substring(line.indexOf(FILE_TEMP_MARKER) + FILE_TEMP_MARKER.length());
+                Integer intTemp = Integer.parseInt(tempString);
+                return BigDecimal.valueOf(intTemp).movePointLeft(3);
+            } else {
+                logger.error(
+                        "GPIO file didn't contain line with 't=' where temperature should be stored. Check if configuration points to the proper file");
+                return INVALID_TEMPERATURE;
+            }
+
+        } catch (IOException e) {
+            logger.error("error reading GPIO bus file. File path is: " + gpioFile + ". Check if path is proper. ", e);
```

Comment:
Please use parametrized logging instead of string concatenation. E.g.
```
logger.error("error reading GPIO bus file. File path is: {}. Check if path is proper. ", gpioFile, e);
```